File: 66 Plus One/66plus-one.py
class Solution(object):
    def plusOne(self, digits):
        """
        :type digits: List[int]
        :rtype: List[int]
        """
        
        l = len(digits)
        carry = 1
        
        for i in range(l-1, -1, -1):
            s = carry + digits[i]
            
            if s > 9:
                carry = 1
                s -= 10
            else:
                carry = 0
                
            digits[i] = s
            
            if i == 0 and carry == 1:
                digits.insert(0, 1)
        
        return digits

File: 66 Plus One/66plus-one.cpp
class Solution {
public:
    vector<int> plusOne(vector<int>& digits) {
        std::vector<int> res;
        int carry = 0;

        for (int i = digits.size() - 1; i >= 0; i--){
            if (i == digits.size() - 1){
                carry = (digits[i] + 1) / 10;
                digits[i] = (digits[i] + 1) % 10;
            } else {
                if (carry != 0){
                    carry = (digits[i] + 1) / 10;
                    digits[i] = (digits[i] + 1) % 10;
                } else {
                    break;
                }
            }
        }

        if (carry == 1) {
            digits.insert(digits.begin(), 1);
        }

        return digits;
    }
};